packages/commonwealth/server/util/cosmosCache.ts

```diff
@@ -1,85 +1,90 @@
+  // Matches ProposalStatus from common-common/src/cosmos-ts/src/codegen/cosmos/gov/v1/gov.ts
+  const activeProposalCodes = [1, 2]; // ['DepositPeriod', 'VotingPeriod']
+  const completedProposalCodes = [3, 4, 5]; // ['Passed', 'Rejected', 'Failed']
+  const url = req.url;
+  const proposalStatus = req?.query?.proposal_status;
+  let duration = defaultCacheDuration;
+
+  if (proposalStatus) {
+    if (activeProposalCodes.some((c) => c === +proposalStatus)) {
       // ACTIVE PROPOSALS: short cache
       duration = oneBlock;
-    } else if (completedCodes.some((c) => c === body?.params?.data)) {
+    } else if (completedProposalCodes.some((c) => c === +proposalStatus)) {
       // COMPLETED PROPOSALS: cache 15 minutes
       duration = 60 * 15;
     }
+    // specific proposal request:
+  } else if (url.includes('/proposals/')) {
+    // LIVE VOTES/TALLY/DEPOSITS: 5 minutes
+    duration = 60 * 5;
+  } else if (url.includes('/params/')) {
+    // PARAMS: cache long term (1 day)
+    duration = 60 * 60 * 24;
+  }
 
-    return cacheDecorator.cacheMiddleware(duration, cosmosRPCKeyGenerator)(
-      req,
-      res,
-      next
-    );
-  };
+  req.cacheDuration = duration;
+  req.cacheKey = req.originalUrl;
+
+  return next();
 };
 
-export const cosmosCacheLCD = (duration: number): RequestHandler => {
-  // Matches ProposalStatus from common-common/src/cosmos-ts/src/codegen/cosmos/gov/v1/gov.ts
-  const activeProposalCodes = [1, 2]; // ['DepositPeriod', 'VotingPeriod']
-  const completedProposalCodes = [3, 4, 5]; // ['Passed', 'Rejected', 'Failed']
+const cosmosRPCDuration = (body, method, params, next) => {
+  let duration = defaultCacheDuration;
+  // RPC specific codes from cosmJS requests
+  const activeCodes = ['0801', '0802']; // ['DepositPerion', 'VotingPeriod']
+  const completedCodes = [
+    '0803', // 'Passed'
+    '0804', // 'Rejected'
+    '0805', // 'Failed'
+    '0803220a0a080000000000000087', // 'Passed - Osmosis'
+    '0803220a0a080000000000000100', // 'Passed - Osmosis'
+    '0803220a0a080000000000000189', // 'Passed - Osmosis'
+  ];
 
-  return async function cacheIfCacheworthyLCD(req, res, next) {
-    const url = req.url;
-    const proposalStatus = req?.query?.proposal_status;
-
-    if (proposalStatus) {
-      if (activeProposalCodes.some((c) => c === +proposalStatus)) {
-        // ACTIVE PROPOSALS: short cache
-        duration = oneBlock;
-      } else if (completedProposalCodes.some((c) => c === +proposalStatus)) {
-        // COMPLETED PROPOSALS: cache 15 minutes
-        duration = 60 * 15;
-      }
-      // specific proposal request:
-    } else if (url.includes('/proposals/')) {
-      // LIVE VOTES/TALLY/DEPOSITS: 5 minutes
-      duration = 60 * 5;
-    } else if (url.includes('/params/')) {
-      // PARAMS: cache long term (1 day)
-      duration = 60 * 60 * 24;
-    }
+  if (/^(tx)/.test(method)) {
+    // TX Response: do not cache
+    return null;
```

Comment:
@mhagel bug i introduced was calling next() from here!
instead returning null now and doing this in calling function
```
req.cacheDuration = cosmosRPCDuration(body, method, params, next);
  if(req.cacheDuration === null) return next();
```